### PR TITLE
executor: fix admin check table bug when the column is nil and has a default value

### PR DESCRIPTION
**Pull request by winkyao**
We can reproduce this bug by:

```
mysql>CREATE TABLE `30951bba-9f99-4342-b928-4b231359726e` (
  `51259c10-0f6f-4b1d-8696-5520f692cab6` int(11) NOT NULL,
  `efde0642-2c17-46d4-9b6e-8ae9e8255699` int(11) NOT NULL,
  PRIMARY KEY (`51259c10-0f6f-4b1d-8696-5520f692cab6`,`efde0642-2c17-46d4-9b6e-8ae9e8255699`),
  KEY `50e13ba5-3333-4195-9346-a9fafa44fad9` (`51259c10-0f6f-4b1d-8696-5520f692cab6`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin
Query OK, 0 rows affected (0.01 sec)

mysql> insert into `30951bba-9f99-4342-b928-4b231359726e`(`51259c10-0f6f-4b1d-8696-5520f692cab6`, `efde0642-2c17-46d4-9b6e-8ae9e8255699`) values(2, 2);                                                                                       Query OK, 1 row affected (0.11 sec)

mysql> ALTER TABLE `30951bba-9f99-4342-b928-4b231359726e` ADD COLUMN `4fa8081c-2b49-46e1-859e-5e43fad70200` int NULL DEFAULT '1795454803' AFTER `51259c10-0f6f-4b1d-8696-5520f692cab6`;
Query OK, 0 rows affected (0.02 sec)

mysql> alter table `30951bba-9f99-4342-b928-4b231359726e` add index b(`4fa8081c-2b49-46e1-859e-5e43fad70200`);
Query OK, 0 rows affected (0.02 sec)

mysql> admin check table `30951bba-9f99-4342-b928-4b231359726e`;
ERROR 1105 (HY000): 30951bba-9f99-4342-b928-4b231359726e err:[admin:1]index:&{1 [{1 0 0 0 1795454803 [] <nil>}]} != record:&{1 [{0 0 0 0 0 [] <nil>}]}
```

This is because we will treat nil column as its default value, and use the default value to backfill index, but admin check table doesn't keep this feature.

### CODE REVIEW

**Comment by shenli** [comment]
@coocood @zimulala PTAL

File: util/admin/admin.go [hunk 1]
```diff
@@ -293,14 +293,29 @@ func checkIndexAndRecord(txn kv.Transaction, t table.Table, idx table.Index) err
...
 
 	startKey := t.RecordKey(0)
 	filterFunc := func(h1 int64, vals1 []types.Datum, cols []*table.Column) (bool, error) {
+		for i, val := range vals1 {
+			col := cols[i]
+			if val.IsNull() {
+				if mysql.HasNotNullFlag(col.Flag) {
+					return false, errors.New("Miss")
+				}
+				// NULL value is regard as its default value.
```
Comment (zhexuany) [inline util/admin/admin.go]: is regarded as?

File: util/admin/admin.go [hunk 2]
```diff
@@ -471,8 +486,17 @@ func rowWithCols(txn kv.Retriever, t table.Table, h int64, cols []*table.Column)
 			continue
 		}
 		ri, ok := row[col.ID]
-		if !ok && mysql.HasNotNullFlag(col.Flag) {
-			return nil, errors.New("Miss")
+		if !ok {
+			if mysql.HasNotNullFlag(col.Flag) {
+				return nil, errors.New("Miss")
+			}
+			// NULL value is regard as its default value.
```
Comment (zhexuany) [inline util/admin/admin.go]: is regarded as?

File: util/admin/admin.go [hunk 3]
```diff
@@ -471,8 +486,17 @@ func rowWithCols(txn kv.Retriever, t table.Table, h int64, cols []*table.Column)
 			continue
 		}
 		ri, ok := row[col.ID]
-		if !ok && mysql.HasNotNullFlag(col.Flag) {
-			return nil, errors.New("Miss")
+		if !ok {
+			if mysql.HasNotNullFlag(col.Flag) {
+				return nil, errors.New("Miss")
+			}
```
Comment (zimulala) [inline util/admin/admin.go]: Do we need to check the column's schema state?

Comment (winkyao) [reply util/admin/admin.go]: checked it in line 481

**Comment by zimulala** [comment]
LGTM

**Comment by winkyao** [comment]
@coocood @ciscoxll PTAL

**Comment by coocood** [comment]
LGTM

**Pull request merged**